src/main/java/net/barrage/school/java/ecatalog/web/MerchantController.java

```diff
@@ -0,0 +1,46 @@
+package net.barrage.school.java.ecatalog.web;
+
+import lombok.extern.slf4j.Slf4j;
+import net.barrage.school.java.ecatalog.app.services.MerchantService;
+import net.barrage.school.java.ecatalog.model.Merchant;
+import net.barrage.school.java.ecatalog.model.Product;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+import java.util.Set;
+
+@Slf4j
+@RestController
+@RequestMapping("/e-catalog/api/v1/merchants")
+public class MerchantController {
+    private final MerchantService merchantService;
+
+    public MerchantController(
+            MerchantService merchantService) {
+        this.merchantService = merchantService;
+    }
+
+    @GetMapping
+    public List<Merchant> listMerchants() {
+        var merchants = merchantService.listMerchants();
+        log.trace("listMerchants -> {}", merchants);
+        return merchants;
+    }
+
+    @GetMapping("/{merchant_id}")
+    public Merchant getMerchantById(@PathVariable("merchant_id") Long merchantId) {
+        var merchant = merchantService.getMerchantById(merchantId);
+        log.trace("getMerchantById -> {}", merchant);
+        return merchant;
+    }
+
+    @GetMapping("/{merchant_id}/products")
+    public Set<Product> listProductsByMerchantId(@PathVariable("merchant_id") Long merchantId) {
+        var products = merchantService.listProductsByMerchant(merchantId);
```

Comment:
imho `listProductsByMerchant` should be a part of ProductService API, not MerchantService